### PR TITLE
[config] refacto

### PR DESCRIPTION
**Pull request by vysahjk**

### CODE REVIEW

File: Babylon/utils/yaml_utils.py [hunk 1]
```diff
@@ -70,6 +96,29 @@ def write_yaml_value(yaml_file: pathlib.Path, keys: Union[list[str], str], value
...
+    :param value: the new value of the key
+    :return : None
+    """
+    _commented_yaml_loader = YAML()
+    try:
+        with yaml_file.open(mode='r') as file:
+            _y = _commented_yaml_loader.load(file) or {}
+            _temp = _y[context_id]
+            _z = set_nested_key(_temp, keys, value)
+            _y[context_id] = _z
```
Comment (enocquet) [inline Babylon/utils/yaml_utils.py:115]: Possible to change naming of vars ?

File: Babylon/utils/yaml_utils.py [hunk 2]
```diff
@@ -116,3 +165,22 @@ def complete_yaml(template_yaml: pathlib.Path, target_yaml: pathlib.Path):
...
+                              resource: str, tenant_id: str):
+    organization_name = os.environ.get('BABYLON_ORG_NAME', '')
+    response = hvac_client.read(path=f'{organization_name}/{tenant_id}/babylon/config/{resource}/{key_name}',)
+    if not response:
+        logger.info(f"{organization_name}/{tenant_id}:babylon/config/{resource}/{key_name} not found")
+        sys.exit(1)
+    data = unflatten_list(response['data'], separator=".")
+    temp = dict()
+    temp[context_id] = data
+    yaml_file = yaml.dump(temp)
```
Comment (enocquet) [inline Babylon/utils/yaml_utils.py:180]: temp is no more used after so maybe shorten it like that : 
```python
yaml_file = yaml.dump({context_id: data})
```

File: Babylon/commands/config/upload.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            logger.info(f"Configuration {environ_id}/{o} successfully created")
+        elif platforms and o in platforms['data']['keys']:
+            logger.info(f"{env.organization_name}/{env.tenant_id}/babylon/config/{environ_id}/{o} already exists")
+            continue
+        else:
+            logger.info(f"New path: [{o}]")
+            data = dict()
+            for i in section_keys:
+                data.update({i: section_data[i]})
+            questions = [inquirer.Text(test, message=test, default=data[test]) for test in section_keys]
```
Comment (enocquet) [inline Babylon/commands/config/upload.py:64]: FIrst line we got: 
```python
default=data['data'][test]
```
and last line:
```python
default=data[test]
```
Just to be sure: not a typo ?

File: Babylon/commands/config/upload.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    rewrite: list,
+    yes: bool,
+) -> CommandResponse:
+    """
+    Register or update a new configuration
+    """
+    yaml_loader = YAML()
+    environ_id = env.environ_id
+    rewrite = list(rewrite)
+    platforms = hvac_client.list(path=f"{env.organization_name}/{env.tenant_id}/babylon/config/{environ_id}")
```
Comment (enocquet) [inline Babylon/commands/config/upload.py:31]:
```python
{env.organization_name}/{env.tenant_id}/babylon/config/{environ_id}
```
is used 6 times in this function... Maybe go for var instead ?

File: Babylon/utils/__init__.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import pathlib
 
 BABYLON_PATH = list(pathlib.Path(__file__).parents)[1]
-TEMPLATE_FOLDER_PATH = BABYLON_PATH / "templates"
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/utils/__init__.py:4]: yes it is

File: Babylon/commands/config/upload.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        question = [inquirer.Confirm(name="new_platform", message="Do you want continue")]
+        response = inquirer.prompt(question)
+        if not response['new_platform']:
+            return CommandResponse.fail()
+
+    for o in config_files:
+        section_path = env.working_dir.original_config_dir / f"{o}.yaml"
+        section_data = yaml_loader.load(section_path.absolute())
+        section_keys = dict(section_data).keys()
+        if platforms and o in rewrite:
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/config/upload.py:42]: maybe: `config_file`

File: Babylon/utils/clients.py [hunk 7]
```diff
@@ -63,12 +66,32 @@ def get_docker_client(registry: str):
...
 
 
+def pass_hvac_client(func: Callable[..., Any]) -> Callable[..., Any]:
+    """Grab hvac client"""
+
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        client = None
+        try:
+            client = hvac.Client(url=env.server_id, token=os.environ.get('BABYLON_TOKEN'))
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/utils/clients.py:76]: yes, there is a check before... in all command group

File: Babylon/utils/credentials.py [hunk 8]
```diff
@@ -1,74 +1,64 @@
...
-    env = Environment()
     credential = None
-    redirect_uri_port = env.configuration.get_platform_var("redirect_uri") or "8842"
-    redirect_uri = f"http://localhost:{redirect_uri_port}"
-
+    babylon_client_id = env.configuration.get_var(resource_id="babylon", var_name="client_id")
     try:
-        credential = EnvironmentCredential(logging_enable=True)
-        if credential._credential is None:
+        baby_client_secret = env.get_env_babylon(name="client")
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/utils/credentials.py:54]: Yes, i think is better for users

File: Babylon/commands/config/upload.py (hunk 3 above)
Comment (vysahjk) [reply Babylon/commands/config/upload.py:64]: nop... i will change name of vars

File: Babylon/utils/credentials.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (enocquet) [reply Babylon/utils/credentials.py:54]: agree

File: Babylon/utils/update_section.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        for i, k in tmp.items():
+            s.update({f"{section}.{i}": k})
+    else:
+        s[section] = new_value
+
+    if unflatten:
+        final = unflatten_list(s, separator=sep)
+    else:
+        final = s
+    return final
```
Comment (enocquet) [inline Babylon/utils/update_section.py:40]: Sorry but those _'one letter vars'_ are terribles ^^
I understand this usage for some cases just like for _enumerate_ for instance, but too many here and that makes the code difficult to read in the end

**Review by enocquet — APPROVED** [review]
Let's go ^^

**Pull request merged**